azurelinuxagent/common/protocol/wire.py

```diff
@@ -790,35 +782,30 @@ def _update_from_goal_state(self, refresh_type):
                 new_goal_state = GoalState(new_goal_state_xml)
 
                 def update_host_plugin():
-                    if self.host_plugin is not None:
-                        self.host_plugin.container_id = new_goal_state.container_id
-                        self.host_plugin.role_config_name = new_goal_state.role_config_name
+                    if self._host_plugin is not None:
+                        self._host_plugin.update_container_id(new_goal_state.container_id)
+                        self._host_plugin.update_role_config_name(new_goal_state.role_config_name)
```

Comment:
Shouldn't you be using the getter for the host plugin here?

Reply:
The getter for HostGAPlugin will create a new object if its not only present. If I call it here then the update call would technically create a new object which was not the behavior previously, I didnt want to diverge a lot from the old code as it might have some unprecedented circumstances. Mainly due to those reasons I just used the property here instead of calling the getter

Reply:
I don't think we could even be in the situation where `update_host_plugin()` is called without the `HostPlugin` having already been initialized. But fair point, it's better to minimize changes for this iteration and do the necessary testing and cleanup later.